presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/StaticMetastoreLocator.java

```diff
@@ -138,34 +152,46 @@ private static URI checkMetastoreUri(URI uri)
         return uri;
     }
 
-    private static class Backoff
+    static class Backoff
     {
-        private static final long MIN_BACKOFF = new Duration(50, MILLISECONDS).roundTo(NANOSECONDS);
-        private static final long MAX_BACKOFF = new Duration(60, SECONDS).roundTo(NANOSECONDS);
+        static final long MIN_BACKOFF = new Duration(50, MILLISECONDS).roundTo(NANOSECONDS);
+        static final long MAX_BACKOFF = new Duration(60, SECONDS).roundTo(NANOSECONDS);
 
-        private final Ticker ticker = Ticker.systemTicker();
+        private final Ticker ticker;
         private long backoffDuration = MIN_BACKOFF;
-        private OptionalLong lastFailureTimestamp = OptionalLong.empty();
+        private boolean success = true;
+        private long lastFailureTimestamp;
+
+        Backoff(Ticker ticker)
+        {
+            this.ticker = requireNonNull(ticker, "ticker is null");
+        }
 
         synchronized void fail()
         {
-            lastFailureTimestamp = OptionalLong.of(ticker.read());
+            success = false;
+            lastFailureTimestamp = ticker.read();
             backoffDuration = min(backoffDuration * 2, MAX_BACKOFF);
         }
 
         synchronized void success()
         {
-            lastFailureTimestamp = OptionalLong.empty();
+            success = true;
```

Comment:
do you want to reset `lastFailureTimestamp` here (as it used to be)?
note that you do not sort over `success` field

Reply:
Reworked to retain `lastFailureTimestamp` as OptionalLong and dropped success flag